### DISCUSSION TITLE
Which version of JCIFS is exposed to rhino scripts?

### DISCUSSION THREAD

**Discussion by why-not-try-calmer** [nextgenhealthcare/connect#6260] [Frequently Asked Questions] [answered]
If I look at the tests in this very repo I see the modern JCIFS API being used, with the two parameter version of the smbFile method. This is the API I want to use, yet in my channels (mirth connect 4.2) I am not able to. The only thing that works is the one-place version of the smbFile method, that takes one of these funny looking urls in the shape of smb//domain;user:password@host...

How to use the more modern API in channels?

**Comment by rjkroll** [comment]


**Reply by tonygermano** [reply]
Your source is not this repo, but an old fork that hasn't been updated in 8 years. Mirth has been using jcifs-ng since version 3.9 from https://github.com/AgNO3/jcifs-ng

Mirth 4.2 and 4.5 both use version 2.1.8
Mirth development branch is using 2.1.10

**Comment by why-not-try-calmer** [comment]
Is that so? Then I don't seem able to use BaseConfiguration, imported as as

`importPackage(Packages.jcifs.config.BaseConfiguration);`

and used as

`var config = new BaseConfiguration();`

as it throws: "BaseConfiguration is an object, not a function".

Am I missing something?

**Reply by tonygermano** [reply, marked as answer]
In mirth 4.5 I get
> Function importPackage must be called with a package; had "[JavaClass jcifs.config.BaseConfiguration]" instead.

I tried this instead:
```javascript
var BaseConfiguration = Packages.jcifs.config.BaseConfiguration
var config = new BaseConfiguration()
```
Which gave me
> Java constructor for "jcifs.config.BaseConfiguration" with arguments "" not found.

And that led me to believe that it was finding the class now, but it does not have a zero-argument constructor.

https://github.com/AgNO3/jcifs-ng/blob/2d5b46907f6614b4d600e96128e0060982e03760/src/main/java/jcifs/config/BaseConfiguration.java#L145-L164

The zero-argument constructor does exist, but it's protected, so you can't call it. It calls the public constructor with a value of false, but I think you really want.
```javascript
var BaseConfiguration = Packages.jcifs.config.BaseConfiguration
var config = new BaseConfiguration(true)
```

**Reply by why-not-try-calmer** [reply]
This clarifies it, thanks a lot!
